web/helpers/state.helper.ts

```diff
@@ -17,3 +17,37 @@ export const sortStates = (states: IState[]) => {
     return Object.keys(STATE_GROUPS).indexOf(stateA.group) - Object.keys(STATE_GROUPS).indexOf(stateB.group);
   });
 };
+
+export const getCurrentStateSequence = (groupSates: IState[], destinationData: any, edge: string | undefined) => {
```

Comment:
Can you change this `any` and define the Type.

Also please add comments where ever possible.